### PR TITLE
feat(kuma-cp): add possibility to split datadog services based on traffic direction and destination

### LINKED ISSUE #4937: Record outbound requests as separate Datadog services
### Description

See https://kuma-mesh.slack.com/archives/CN2GN4HE1/p1661205593905379. Currently, ingress into a service and egress are recorded on the same Datadog service as different resources, which can be confusing. Would like to add the option to separate them into different ingress and egress Datadog services.

### PR DESCRIPTION
**Pull request by Automaat**
add possibility to split datadog services based on traffic direction and destination

Fix #4937

[Docs pr](https://github.com/kumahq/kuma-website/pull/1012)

Signed-off-by: Marcin Skalski <marcin.skalski@konghq.com>

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: api/mesh/v1alpha1/mesh.proto [hunk 1]
```diff
@@ -199,6 +199,10 @@ message DatadogTracingBackendConfig {
 
   // Port of datadog collector
   uint32 port = 2 [ (doc.required) = true ];
+
+  // Determines if datadog service name should be split based on traffic
+  // direction and destination. Default: false
```
Comment (jakubdyszkiewicz) [inline api/mesh/v1alpha1/mesh.proto:204]: How about we include an example of what the names look like if this is true?

**Review by jakubdyszkiewicz — COMMENTED** [review]
nice!

File: api/mesh/v1alpha1/mesh.proto (hunk 1 above)
Comment (kleinfreund) [reply api/mesh/v1alpha1/mesh.proto:204]: Like the ones in https://github.com/kumahq/kuma-website/pull/1012.

**Comment by Automaat** [comment]
I am updating the screenshot of how this split looks right now in datadog. I think that this split looks a little bit strange, but I understand that in some cases you can build more precise statistics from this data

Without split:
Screenshot 2022-09-27 at 12 07 25

With services split:
Screenshot 2022-09-27 at 12 08 01

**Comment by Automaat** [comment]
I have one question, when setting up datadog agent I followed the instruction from Kuma docs. I created k8s service for datadog agent like this:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: trace-svc
spec:
  selector:
    app: datadog
  ports:
    - protocol: TCP
      port: 8126
      targetPort: 8126
```

and then I created datadog backend as it docs:
```yaml
apiVersion: kuma.io/v1alpha1
kind: Mesh
metadata:
  name: default
spec:
  tracing:
    defaultBackend: datadog-collector
    backends:
    - name: datadog-collector
      type: datadog
      sampling: 100.0
      conf:
        address: trace-svc.datadog.svc.cluster.local
        port: 8126
```

But envoy was unable to connect to: `trace-svc.datadog.svc.cluster.local`. Did I do something wrong? 

Edit. While writing this comment, I get what is wrong. This works, but this assumes that Datadog agent is in `datadog` namespace and has label `app=datog` should we update Datadog setup docs in Kuma accordingly?

**Comment by jakubdyszkiewicz** [comment]
> should we update Datadog setup docs in Kuma accordingly?

yes please

**Review by jakubdyszkiewicz — APPROVED** [review]

File: api/mesh/v1alpha1/mesh.proto (hunk 1 above)
Comment (Automaat) [reply api/mesh/v1alpha1/mesh.proto]: updated

**Comment by lahabana** [comment]
> Edit. While writing this comment, I get what is wrong. This works, but this assumes that Datadog agent is in datadog namespace and has label app=datog. When following Datadog docs, datadog agent is created in default namespace, should we update Datadog setup docs in Kuma accordingly?

@Automaat was this fixed or have we at least created an issue in the docs? Thx

**Pull request merged**